cs/src/core/VarLen/MemoryFunctions.cs

```diff
@@ -11,17 +11,15 @@ public class MemoryFunctions<Key, T, Context> : FunctionsBase<Key, Memory<T>, Me
         where T : unmanaged
     {
         readonly MemoryPool<T> memoryPool;
-        readonly bool locking;
 
         /// <summary>
         /// Constructor
         /// </summary>
         /// <param name="memoryPool"></param>
         /// <param name="locking">Whether we lock values before concurrent operations (implemented using a spin lock on length header bit)</param>
-        public MemoryFunctions(MemoryPool<T> memoryPool = default, bool locking = false)
+        public MemoryFunctions(MemoryPool<T> memoryPool = default, bool locking = false) : base(locking)
```

Comment:
Why are we still doing locking inside ConcurrentWriter below? Shouldn't it be refactored into Lock() and Unlock() overrides.